codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/localservice/extensions/DafnyPythonLocalServiceStructureGenerator.java

```diff
@@ -203,7 +309,14 @@ protected void writePropertyForMember(
       );
     }
 
-    if (target.hasTrait(ReferenceTrait.class)) {
+    // Reference shapes require forward reference to avoid circular import,
+    // but references to AWS SDKs don't
```

Comment:
Nit picking, but isn't it more accurately that shapes from dependencies are already declared?

Reply:
This is actually about circular dependencies within one project.
I'm going to write about this problem and solutions in a Smithy-Dafny Python dev guide, but an overview/instance of the problem is that `models.py` can't import `references.py` at the top-level because `references.py` depends on `models.py` at the top-level, but `models.py` still needs to import `references.py` to use its classes.
One solution is to defer the imports like I've done here.
I wrote a fair amount about this and drew some graphs, but I'm going to collate all that later.

Reply:
I will be thrilled to review a PR that targets `docs/python` in the future :)